File: languages/java/app-encryption/src/main/java/com/godaddy/asherah/crypto/NeverExpiredCryptoPolicy.java
package com.godaddy.asherah.crypto;

import java.time.Instant;

public class NeverExpiredCryptoPolicy implements CryptoPolicy {

  @Override
  public boolean isKeyExpired(final Instant keyCreationDate) {
    return false;
  }

  @Override
  public long getRevokeCheckPeriodMillis() {
    return Long.MAX_VALUE;
  }

  @Override
  public boolean canCacheSystemKeys() {
    return true;
  }

  @Override
  public boolean canCacheIntermediateKeys() {
    return true;
  }

  @Override
  public boolean useSharedIntermediateKeyCache() {
    return false;
  }

  @Override
  public long getSharedIkCacheExpireAfterAccessMillis() {
    return Long.MAX_VALUE;
  }

  @Override
  public boolean notifyExpiredIntermediateKeyOnRead() {
    return true;
  }

  @Override
  public boolean notifyExpiredSystemKeyOnRead() {
    return true;
  }

  @Override
  public KeyRotationStrategy keyRotationStrategy() {
    return KeyRotationStrategy.INLINE;
  }

}

File: csharp/AppEncryption/Crypto/NeverExpiredCryptoPolicy.cs
using System;

namespace GoDaddy.Asherah.Crypto
{
  public class NeverExpiredCryptoPolicy : CryptoPolicy
  {
    /// <inheritdoc />
    public override bool IsKeyExpired(DateTimeOffset keyCreationDate)
    {
      return false;
    }

    /// <inheritdoc />
    public override long GetRevokeCheckPeriodMillis()
    {
      return long.MaxValue;
    }

    /// <inheritdoc />
    public override bool CanCacheSystemKeys()
    {
      return true;
    }

    /// <inheritdoc />
    public override bool CanCacheIntermediateKeys()
    {
      return true;
    }

    /// <inheritdoc />
    public override bool CanCacheSessions()
    {
      return false;
    }

    /// <inheritdoc />
    public override long GetSessionCacheMaxSize()
    {
      return long.MaxValue;
    }

    /// <inheritdoc />
    public override long GetSessionCacheExpireMillis()
    {
      return long.MaxValue;
    }

    /// <inheritdoc />
    public override bool NotifyExpiredIntermediateKeyOnRead()
    {
      return true;
    }

    /// <inheritdoc />
    public override bool NotifyExpiredSystemKeyOnRead()
    {
      return true;
    }

    /// <inheritdoc />
    public override KeyRotationStrategy GetKeyRotationStrategy()
    {
      return KeyRotationStrategy.Inline;
    }
  }
}
